models/articles.model.js

```diff
@@ -10,14 +11,23 @@ exports.selectArticles = () => {
 
 exports.selectArticleById = (article_id) => {
 	const queryTxt = format('SELECT * FROM articles WHERE article_id = $1');
-	return db.query(queryTxt, [article_id]).then((result) => {
-		if (result.rows.length === 0) {
-			return Promise.reject({ status: 404, msg: 'Not found' });
-			// return the promise
-		} else {
-			return result.rows[0];
-		}
-	});
+	const commentCountQueryTxt = format('SELECT * FROM comments WHERE article_id=$1');
+
+	return db
+		.query(queryTxt, [article_id])
+		.then((result) => {
+			if (result.rows.length === 0) {
+				return Promise.reject({ status: 404, msg: 'Not found' });
+			}
+			return result;
+		})
+		.then((result) => {
+			return db.query(commentCountQueryTxt, [article_id]).then((commentById) => {
```

Comment:
You don't need to do two queries here. There's a much more succinct way of getting the comment count.